### PR TITLE
UDF extensions part one

### PR DESCRIPTION
**Pull request by hannesmuehleisen**
- UDFs can now have 0, 1, 2, or 3 arguments
- Arguments can be `NULL`, return values can be `NULL`
- `VARCHAR` and `DOUBLE` arguments and return values, also `TINYINT`/`SMALLINT`/`INTEGER` and their unsigned versions

### CODE REVIEW

File: packages/duckdb-wasm/src/bindings/udf_runtime.ts [hunk 1]
```diff
@@ -14,36 +14,212 @@ function storeError(mod: DuckDBModule, response: number, message: string) {
...
+        }
+
+        // schema description as json
+        const desc_str = new TextDecoder().decode(mod.HEAPU8.subarray(descPtr, descPtr + descSize));
+        const desc = JSON.parse(desc_str);
+        //console.log(desc);
+
+        // array of buffer pointers referred to from schema
+        const ptrs_buf = mod.HEAPU8.subarray(ptrsPtr, ptrsPtr + ptrsSize);
+        const ptrs = new BigUint64Array(ptrs_buf.buffer, ptrs_buf.byteOffset, ptrsSize / 8);
```
Comment (ankoh) [inline packages/duckdb-wasm/src/bindings/udf_runtime.ts:93]: We might want to change this one later.
https://caniuse.com/?search=BigUint64Array

We do a platform check for BigIntArrays in the remaining API.
https://github.com/duckdb/duckdb-wasm/blob/d050191e1cad0525b40d466d714610543371739f/packages/duckdb-wasm/src/platform.ts#L67

**Comment by ankoh** [comment]
Awesome, thx 👍

**Review by ankoh — APPROVED** [review]

File: packages/duckdb-wasm/src/bindings/udf_runtime.ts (hunk 1 above)
Comment (hannesmuehleisen) [reply packages/duckdb-wasm/src/bindings/udf_runtime.ts:93]: Ah missed that one, will just switch them to Double, too, seems to work fine

**Comment by Y--** [comment]
Awesome! Great work :-)

**Pull request merged**